tests/nested/core20/kernel-reseal/task.yaml

```
@@ -2,19 +2,37 @@ summary: Check that a kernel refresh reseals
 
 systems: [ubuntu-20.04-64]
 
-execute: |
+prepare: |
   # shellcheck source=tests/lib/nested.sh
   . "$TESTSLIB/nested.sh"
 
-  # Wait for snapd to be seeded
-  nested_exec sudo snap wait system seed.loaded
+  snap download pc-kernel --channel=20/edge --basename=pc-kernel
+  unsquashfs -d pc-kernel pc-kernel.snap
+  # ensure we really have the header we expect
+  grep -q -a "This program cannot be run in DOS mode" pc-kernel/kernel.efi
+  # modify the kernel so that the hash changes
+  sed -i 's/This program cannot be run in DOS mode/This program cannot be run in D0S mode/' pc-kernel/kernel.efi
+
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+  nested_secboot_sign_file "$SNAKEOIL_KEY" "$SNAKEOIL_CERT" pc-kernel/kernel.efi
```

ah sorry I missed this in my review, but this is wrong, it should be like this

```suggestion
  nested_secboot_sign_file "$PWD/pc-kernel" "$SNAKEOIL_KEY" "$SNAKEOIL_CERT" kernel.efi
```